### PR TITLE
[Blogging Reminders Sync] Implement blogging reminders resolver

### LINKED ISSUE #17279: [Blogging Reminders Sync] Implement blogging reminders resolver

### PR DESCRIPTION
**Pull request by RenanLukas**
Fixes #17279

To test:
1 - Install WordPress and log in;
2 - Add blogging prompts to a couple of different sites;
3 - Set the flag `isFeatureFlagEnabled` to `true` on `SharedLoginResolver.kt`, `UserFlagsResolver.kt` **and** `BloggingRemindersResolver.kt` (this cannot be done using the `Debug settings` screen because that is only available when the user is already logged in);
4 - Replace `jetpackProviderSyncFeatureConfig.isEnabled()` with `true` on `SharedLoginProvider.kt`, `UserFlagsProvider.kt` **and** `BloggingRemindersProvider.kt`.
5 - Install Jetpack using the same variant (e.g. if installed WordPress version was `jalapenoDebug`, install Jetpack `jalapenoDebug`);
6 - Jetpack should login automatically after launching, and home screen should be shown;
7 - Check if the blogging prompts set on step 2 were set correctly in the site settings.

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and unit tests.

3. What automated tests I added (or what prevented me from doing so)
Implemented/updated: `BloggingRemindersSyncAnalyticsTrackerTest.kt`, `JetpackBloggingRemindersSyncFlagTest.kt`, `BloggingRemindersResolverTest.kt`, `QueryResultTest.kt` and `ClientVerificationTest.kt`.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/build.gradle [hunk 1]
```diff
@@ -122,6 +122,7 @@ android {
         buildConfigField "boolean", "JETPACK_POWERED_BOTTOM_SHEET", "false"
         buildConfigField "boolean", "JETPACK_SHARED_LOGIN", "false"
         buildConfigField "boolean", "JETPACK_LOCAL_USER_FLAGS", "false"
+        buildConfigField "boolean", "JETPACK_BLOGGING_REMINDERS_SYNC_REMOTE_FIELD", "false"
```
Comment (develric) [inline WordPress/build.gradle]: np: maybe we can simplify with JETPACK_BLOGGING_REMINDERS_SYNC.

File: WordPress/src/main/java/org/wordpress/android/bloggingreminders/resolver/BloggingRemindersResolver.kt [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        val bloggingRemindersResultCursor = getBloggingRemindersSyncResultCursor()
+        if (bloggingRemindersResultCursor != null) {
+            val siteModelBloggingReminderMap = queryResult.getValue<SiteIDBloggingReminderMap>(
+                        bloggingRemindersResultCursor, object : TypeToken<SiteIDBloggingReminderMap?>() {}.type
+            ) ?: emptyMap()
+            if (siteModelBloggingReminderMap.isNotEmpty()) {
+                val success = syncBloggingReminders(siteModelBloggingReminderMap)
+                if (success) onSuccess() else onFailure()
+            } else {
+                bloggingRemindersSyncAnalyticsTracker.trackFailed(ErrorType.NoBloggingRemindersFoundError)
```
Comment (develric) [inline WordPress/src/main/java/org/wordpress/android/bloggingreminders/resolver/BloggingRemindersResolver.kt]: I think not finding any reminders is actually a valid use case more than an error. If so we should track success (we may want to track number of reminders synched, with 0 as a valid case) and invoke onSuccess(). Wdyt?

File: WordPress/src/main/java/org/wordpress/android/bloggingreminders/provider/BloggingRemindersProvider.kt [hunk 3]
```diff
@@ -47,13 +51,9 @@ class BloggingRemindersProvider : QueryContentProvider() {
                             bloggingRemindersModel.enabledDays.isNotEmpty()
                         }
                         val filteredSiteIds = filteredBloggingReminders.map { bloggingReminder ->
-                            bloggingReminder.siteId
-                        }
-                        val filteredSiteModels = allSiteModels.filter { siteModel ->
-                            filteredSiteIds.contains(siteModel.id)
+                            siteStore.getSiteIdForLocalId(bloggingReminder.siteId)
```
Comment (develric) [inline WordPress/src/main/java/org/wordpress/android/bloggingreminders/provider/BloggingRemindersProvider.kt:54]: np: I would consider renaiming like `getSiteRemoteIdForLocalId `

File: WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java [hunk 4]
```diff
@@ -1609,6 +1612,9 @@ public void onSiteChanged(OnSiteChanged event) {
                 mSelectedSiteRepository.updateSite(site);
             }
         }
+        mBloggingRemindersResolver.trySyncBloggingReminders(
+                () -> Unit.INSTANCE, () -> Unit.INSTANCE
+        );
```
Comment (develric) [inline WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java:1617]: As we discussed on slack, we are keeping this entry in the onSiteChanged for this PR, but we are going to review this approach possibly in favor of chaining the call with the others sync calls in SharedLoginResolver

File: WordPress/src/main/java/org/wordpress/android/bloggingreminders/resolver/BloggingRemindersResolver.kt [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        val wordpressBloggingRemindersSyncUriValue =
+                "content://${wordPressPublicData.currentPackageId()}.${BloggingRemindersProvider::class.simpleName}"
+        return contentResolverWrapper.queryUri(
+                contextProvider.getContext().contentResolver,
+                wordpressBloggingRemindersSyncUriValue
+        )
+    }
+
+    @Suppress("TooGenericExceptionCaught", "SwallowedException")
+    private fun syncBloggingReminders(siteIDBloggingReminderMap: SiteIDBloggingReminderMap): Boolean {
```
Comment (develric) [inline WordPress/src/main/java/org/wordpress/android/bloggingreminders/resolver/BloggingRemindersResolver.kt]: np: should it be siteIdBloggingReminderMap with lower case d?

**Review by develric — COMMENTED** [review]
Hey @RenanLukas 👋 ; thanks for the PR 🙇 

I did a first pass and left a few comments (mostly np). Maybe the one we should consider more is [this one](https://github.com/wordpress-mobile/WordPress-Android/pull/17308#discussion_r995157344)

Let me know. I'll continue testing tomorrow 🙇

File: WordPress/src/main/java/org/wordpress/android/bloggingreminders/provider/BloggingRemindersProvider.kt (hunk 3 above)
Comment (RenanLukas) [reply WordPress/src/main/java/org/wordpress/android/bloggingreminders/provider/BloggingRemindersProvider.kt:54]: I completely agree, but this method is part of the FluxC library 😕 
IMHO it's worth to add the prefix `remote` or `local` everywhere it says only `siteId`

**Comment by RenanLukas** [comment]
Hey @develric , thanks for the review

Could you please have another look when you can?

**Review by develric — APPROVED** [review]
Thanks for the follow up changes @RenanLukas 🙇 ! Completed testing and it worked as per steps 👍 ! :shipit:

**Pull request merged**